docs/assembly.rst

```diff
@@ -906,8 +940,16 @@ Pseudocode::
 Opcode Stream Generation
 ------------------------
 
-During opcode stream generation, we keep track of the current stack height,
-so that accessing stack variables by name is possible.
+During opcode stream generation, we keep track of the current stack height
+in a counter,
+so that accessing stack variables by name is possible. The stack height is modified with every opcode
+that modifies the stack and with every label that is annotated with a stack
+adjustment. Every time a new
+local variable is introduced, it is registered together with the current
+stack height. If a variable is accessed (either for copying its value or for
```

Comment:
Which sounds more correct?
* a new local variable is introduced on top of the existing local variables, and the new local variable is associated with `the location of the most recent existing local variable + 1`
* a new local variable is introduced on top of the stack, and the new local variable is associated with the current stack height.

Now the description reads like the latter, but I think I heard the former.